temperature/router.py

```diff
@@ -0,0 +1,48 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+from datetime import datetime
+
+from . import models, schemas, crud
+from dependencies import get_db
+
+router = APIRouter()
+
+
+@router.post("/temperatures/update")
+async def update_temperatures(db: AsyncSession = Depends(get_db)):
+    # Get all cities from the database
+    result = await db.execute(select(models.City))
+    cities = result.scalars().all()
+
+    temperatures_data = []
+
+    # Fetch and store temperature for each city
+    for city in cities:
+        try:
+            temperature = await crud.fetch_current_temperature(city.name)
+            temperatures_data.append({
+                "city_id": city.id,
+                "date_time": datetime.now(),
+                "temperature": temperature
+            })
+        except Exception as e:
+            continue  # Skip if fetching temperature fails
```

Comment:
Consider logging the exception `e` for debugging purposes instead of silently continuing. This can help identify issues with fetching temperatures.